src/services/OswServiceTaskWiFi.cpp

```diff
@@ -42,23 +42,53 @@ void OswServiceTaskWiFi::loop() {
     }
 
     if(this->m_enableClient) {
-        if(this->m_autoAPTimeout and WiFi.status() == WL_CONNECTED) {
-            //Nice - reset timeout
-            this->m_autoAPTimeout = 0;
-        } else if(!this->m_autoAPTimeout and WiFi.status() != WL_CONNECTED) {
-            //Wifi is either disconnected or unavailable -> start timeout until we start our own ap
-            this->m_autoAPTimeout = time(nullptr);
+        if(this->m_connectTimeout and WiFi.status() == WL_CONNECTED) {
+            // Nice - reset timeout
+            this->m_connectTimeout = 0;
+            this->m_connectFailureCount = 0;
+        } else if(!this->m_connectTimeout and WiFi.status() != WL_CONNECTED) {
+            // Wifi is either disconnected or unavailable...
+            if(this->onlyOneModeSimultaneously and this->m_enableStation) {
+                // This watch does not support connecting to networks with an active AutoAP.
+                // It would therefore be not helpful to activate the timeout to try an other
+                // action, while the AutoAP is active as this will shutdown the AutoAP as the
+                // timeout expires!
```

Comment:
However, it seems that it will be possible to implement it later. I remember features implemented in a similar model. I don't know where it'll be used :smile:

Reply:
Hmm, maybe scanning the network would help... This could be an alternative solution - but not now :)